pkg/mdformatter/linktransformer/validator.go

```diff
@@ -34,7 +34,11 @@ func (v GitHubValidator) IsValid(k futureKey, r *validator) (bool, error) {
 // RoundTripValidator.IsValid returns true if url is checked by colly.
 func (v RoundTripValidator) IsValid(k futureKey, r *validator) (bool, error) {
 	// Result will be in future.
-	r.destFutures[k].resultFn = func() error { return r.remoteLinks[k.dest] }
+	prevResult, _ := r.destFutures.LoadAndDelete(k)
```

Comment:
Lot's of races when we delete, unfortunately.. we need to make this operation locked potentially with Mutex. 

But it has to be the full operation Locked:
1. Checking if there is entry in map
2. If no, create new future, and schedule colly
3. If yes, grab reference to future

We might need to check `IsValid` interface, if this actually allows us to do this 🤔